tests/v3/index.test.ts

```diff
@@ -988,4 +988,154 @@ describe("OpenAPI3 features", () => {
+                schema: { type: "integer" },
+              },
+            ],
+            responses: {
+              "200": {
+                content: {
+                  "application/json": {
+                    schema: {
+                      type: "object",
+                      properties: {
+                        results: {
+                          type: "array",
+                          items: { $ref: "#/components/schemas/SearchResult" },
+                        },
+                        total: { type: "integer" },
+                      },
+                      required: ["total"],
+                    },
+                  },
+                },
+              },
+              "404": {
+                content: {
+                  "application/json": {
+                    schema: { $ref: "#/components/schemas/ErrorResponse" },
+                  },
+                },
+              },
+            },
+          },
+        },
+      },
+      components: {
+        schemas: {
+          ErrorResponse: {
+            type: "object",
+            properties: {
+              error: { type: "string" },
+              message: { type: "string" },
+            },
+            required: ["error", "message"],
+          },
+          SearchResponse: {
+            type: "object",
+            properties: {
+              title: { type: "string" },
+              date: { type: "string" },
+            },
+            required: ["title", "date"],
+          },
+        },
+        responses: {
+          NotFound: {
+            content: {
+              "application/json": {
+                schema: { $ref: "#/components/schemas/ErrorResponse" },
+              },
+            },
+          },
+        },
+      },
+    };
+
+    expect(swaggerToTS(schema)).toBe(
+      format(`
+      export interface paths {
+        '/': {
+          get: {
+            responses: {
+              '200': {
+                'application/json': { title: string; body: string }
+              }
+            }
+          }
+        };
+        '/search': {
+          post: {
+            parameters: {
+              query: {
+                q: string;
+                p?: number;
+              }
+            };
+            responses: {
+              '200': {
+                'application/json': {
+                  results?: components['schemas']['SearchResult'][];
+                  total: number;
+                }
+              }
+              '404': {
+                'application/json': components['schemas']['ErrorResponse']
+              }
+            }
+          }
+        }
+      }
+
+      export interface operations {}
+        
+      export interface components {
+        schemas: {
+          ErrorResponse: { error: string; message: string };
+          SearchResponse: { title: string; date: string }
+        }
+        responses: {
+          NotFound: { [key: string]: any }
+        }
+      }`)
+    );
+  });
```

Comment:
can you reduce the test case here? Having only the `/` path should be sufficient, you can remove the `/search` bits?

Reply:
[Done!](https://github.com/drwpow/openapi-typescript/pull/366/files/9b31327b90c39d899b788a974e0f7f80d6f045fa..efed10f2a66e1cf1385de147ed01b784e8995b72#diff-f6cd969653f24a9df9131814e4643dc857e0044c9d415cd41445030dafda1e25)

I added a "description" check to 'get' for completeness